File: competitions/onlinejudge/ch2/linear/11850/p11850.py
import sys



stdin = sys.stdin


TOTAL_MILES = 1422


def solution(stations, total_miles=TOTAL_MILES):
    for i in range(len(stations) - 1):
        diff = stations[i+1] - stations[i]
        if diff <= 200:
            continue
        else:
            return False
    if abs(stations[-1] - TOTAL_MILES) * 2 > 200:
        return False
    return True




if __name__ == '__main__':

    while True:
        n = stdin.readline().strip()
        if n == '0':
            break
        n = int(n)
        stations = [0] * n
        for i in range(n):
            stations[i] = int(stdin.readline().strip())

        stations = sorted(stations)
        if solution(stations):
            print('POSSIBLE')
        else:
            print('IMPOSSIBLE')

File: competitions/onlinejudge/ch2/linear/11850/p11850.c
#include<stdio.h>
#include <stdlib.h>

#define TOTAL_MILES 1422

int cmpfunc (const void * a, const void * b) {
   return ( *(short int*)a - *(short int*)b );
}

int main() {
    short int n;
    while (1) {
        scanf("%hd", &n);
        if (n == 0) break;
        short int st[n];
        for(short int i = 0; i<n; i++)
            scanf("%hd", &st[i]);

        char possible = 1;
        qsort(st, n, sizeof(short int), cmpfunc);
        for(short int i = 0; i< n -1; i++) {
            short int diff = st[i+1] - st[i];

            if (diff <= 200) {
                continue;
            }
            else possible = 0;
        }
        if (possible == 0 || ((TOTAL_MILES - st[n-1])*2 > 200)) {
           printf("IMPOSSIBLE\n");
        } else {
           printf("POSSIBLE\n");
        }
    }

    return 0;
}
